sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2018,6 +2018,8 @@ class Analyzer(
             fail(child, DateType, walkedTypePath)
           case (StringType, to: NumericType) =>
             fail(child, to, walkedTypePath)
+          case (from: ArrayType, to: ArrayType) if !from.containsNull =>
```

Comment:
I mean MapType. It's similar to ArrayType, the value of it can be nullable or non-nullable.